Plug modal window

Dima-Penzev opened the pull request.

Dima-Penzev committed:
подключил модалку

Dima-Penzev committed:
fixed modal window

ialiev685 reviewed (changes requested):
<p class="modal__flm-data">
                Popularity <span class="modal-flm-popularity">{{popularity}}</span>
   </p>
Популярность округли пожалуйста до 10-х, данные приходят в 1000-х. Я понимаю делала Света, раз ты подключил, добей пожалуйста.

Dima-Penzev committed:
created function for localStorage

ialiev685 commented on src/js/add-to-watched-btn.js:
```diff
@@ -0,0 +1,37 @@
+
+let num = localStorage.length;
+
+window.addEventListener('click', e => {
```
Зачем вешаешь всегда на window? Можно с карточки всплытие ловить. Иначе пользователь жмакает вне карточки, обработчик обрабатывает любой клик. Считаю это неправильно.

ialiev685 commented on src/js/add-to-watched-btn.js:
```diff
@@ -0,0 +1,37 @@
+  const data = await getTrendItems(page);
+    const properFilm = data.results.find(elem => elem.id === Number(cardNum));
+    
+    localStorage.setItem(`watched-${filmNum}`, JSON.stringify(properFilm));
```
Можно бесконечно добавлять в библиотеку один и тот же фильм.  При добавлении фильма в просмотренные, нужно чтобы текст add to watched менялся соответствующий текст.

ialiev685 commented on src/js/library-btn.js:
```diff
@@ -14,7 +15,7 @@ function openLibrary() {
   refs.homeBtn.classList.remove('current');
   refs.navContainer.style.marginBottom = '49px';
-  
+  getFromLocalStorage();
```
Рендер следующей задачей допилишь?

Dima-Penzev committed:
rounded

Dima-Penzev committed:
complited modal window

ialiev685 commented on src/js/about-btn.js:
```diff
@@ -0,0 +1,35 @@
+});
+
+async function modalFilm(page, cardNum) {
+  const data = await getTrendItems(page);
```
Попробуй в поиске введи слово Rembo и ты увидишь что этот фильм не откроется в модалке. Потому что ты  сравниваешь id фильма Rembo,  c общим запросом по рейтингу. К тебе приходит общий запрос и ты начинаешь искать там id Rembo,  а там его нет. Я на планерке говорил что этот подход не очень, делать многочисленные запросы. Если бы мы все хранили в карточке, было бы проще писать код, не было этих спагети.

ialiev685 commented on src/js/add-to-queue-btn.js:
```diff
@@ -0,0 +1,44 @@
+
+async function addToQueue(page, cardNum) {
+  const data = await getTrendItems(page);
+  const properFilm = data.results.find(elem => elem.id === Number(cardNum));
```
Тоже самое. В хранилище приходит undefined

ialiev685 commented on src/js/add-to-watched-btn.js:
```diff
@@ -0,0 +1,44 @@
+
+async function addToWatched(page, cardNum) {
+  const data = await getTrendItems(page);
+  const properFilm = data.results.find(elem => elem.id === Number(cardNum));
```
Тоже самое. В хранилище приходит undefined. C общим запросом не работает.

ialiev685 commented on src/js/add-to-watched-btn.js:
```diff
@@ -0,0 +1,44 @@
+
+function putProperFilm(params, film) {
+  if (params) {
+    return alert('Этот фильм уже есть в списке просмотренных');
```
Не мешало бы и текст в кнопках менять если уже есть в хранилище.

ialiev685 commented on src/js/add-to-queue-btn.js:
```diff
@@ -0,0 +1,44 @@
+
+function putProperFilm(params, film) {
+    if (params) {
+        return alert('Этот фильм уже есть в очереди');
```
Не мешало бы и текст в кнопках менять если уже есть в хранилище.

JuliaZaykaS committed:
рефакторинг ч1

сняла слушатель с виндоу по закрытию модалки, сделала отдельный запрос за инфой по фильму, вынесла отдельно функцию по рендеру карточки
нужно исправить шаблон, слетела подстановка данных

JuliaZaykaS committed:
доделала рендер модалки

JuliaZaykaS committed:
допилила модалку

JuliaZaykaS committed:
закомментила лишнее

оставила только открытие модалки по кнопке эбаут и ее закрытие

JuliaZaykaS committed:
Merge branch 'main' into plug-modal-window

JuliaZaykaS committed:
связала 2 модалки на открытие/закрытие

JuliaZaykaS committed:
удалила лишние комментарии
